Lib/zipimport.py

```diff
@@ -517,6 +517,7 @@ def _read_directory(archive):
                             num_extra_values = (len(extra_data) - 4) // 8
                             if num_extra_values > 3:
                                 raise ZipImportError(f"can't read header extra: {archive!r}", path=archive)
+                            import struct
```

Comment:
why not import it at the top with the other imports? is it an expensive import?

Reply:
We crossed comments here. No, not individually. On my machine its ~300us. That said, in Pex, I have to fight to shave off the ~O(50ms) boot overhead I get from the pure stdlib imports I use there; so I'm very sensitive to adding more latency to Python application boot times; especially when the addition is only needed for semi-exotic apps that hit these size limits. As I said in the other comment though, just let me know if this should be a standard import and I'll fix.